tests/unit/samples/wiringstringids/resourceclosing.py

```diff
@@ -25,17 +32,49 @@ def __init__(self, service: Service):
         self.service = service
 
 
+class NestedService:
+    def __init__(self, factory_service: FactoryService):
+        self.factory_service = factory_service
+
+
 def init_service():
     service = Service()
     service.init()
     yield service
     service.shutdown()
 
 
+def init_service_with_singleton(singleton: Singleton):
+    service = Service()
+    service.init(singleton)
+    yield service
+    service.shutdown()
+
+
 class Container(containers.DeclarativeContainer):
 
     service = providers.Resource(init_service)
     factory_service = providers.Factory(FactoryService, service)
+    factory_service_kwargs = providers.Factory(
+        FactoryService,
+        service=service
+    )
+    nested_service = providers.Factory(NestedService, factory_service)
+
+
+class ContainerSingleton(containers.DeclarativeContainer):
```

Comment:
Added another container to cover a case where a resource is not in the end of the dependency graph. A `Singleton` provider type is arbitrary - its type doesn't play any role in injections, just needed something for a `Resource` to depend on.